File: gems/decomplex/examples/zig/temporal-ordering-pressure.zig
const TemporalOrderExample = struct {
    a: i32,
    b: i32,

    pub fn one(self: *TemporalOrderExample) void {
        self.a = 1;
    }

    pub fn two(self: *TemporalOrderExample) void {
        self.a = 2;
        self.b = 3;
    }

    pub fn three(self: *TemporalOrderExample) void {
        self.b = 4;
    }

    pub fn reader(self: *TemporalOrderExample) i32 {
        return self.a;
    }
};

File: gems/decomplex/examples/c/temporal-ordering-pressure.c
typedef struct TemporalOrderExample { int a; int b; } TemporalOrderExample;
void one(TemporalOrderExample *self) { self->a = 1; }
void two(TemporalOrderExample *self) { self->a = 2; self->b = 3; }
void three(TemporalOrderExample *self) { self->b = 4; }
int reader(TemporalOrderExample *self) { return self->a; }
